### ISSUE TITLE
Users.GetAllAsync does not return UserMetadata

### ISSUE THREAD

**Issue by jesse10581** [auth0/auth0.net#360]
Hi, I just started using the Auth0 management API and figured I would start with your library. I am having an issue with missing field data.

I am executing Users.GetAllAsync(...) with no field filters and a page num 0 and page size 100. I am able to see the list of users, but the UserMetadata is always null.

If I send a request directly to the API using postman, I see the user_metadata filled in. I think there might be a problem in the PagedListConverter. For now, I will probably just have to use the API manually with HttpClient.

**Comment by jesse10581** [comment]
Hi, I just started using the Auth0 management API and figured I would start with your library.  I am executing Users.GetAllAsync(...) with a page num 0 and page size 100.  I am able to see the list of users, but the UserMetadata is always null.  

If I send a request directly to the API using postman, I see the user_metadata filled in.  I think there might be a problem in the PagedListConverter.  For now, I will probably just have to use the API manually with HttpClient.

**Comment by damieng** [comment]
I just tried a quick test where I added a call to GetAllAsync from our integration test that adds metadata and could not reproduce this problem.

Additional line and debugger showing fully populated UserMetadata using only a sort to make sure I get the most recent user added from the test:

![image](https://user-images.githubusercontent.com/118951/72760341-49a1f600-3b8d-11ea-8e6e-9acbe13acbec.png)

**Comment by jesse10581** [comment]
Was this using the current version of the Nuget package (5.6.5)?  Also I
should have mentioned I am using .Net core 3.1.

**Comment by damieng** [comment]
I tried it using both beta 7 and the latest non-beta which is 6.5.6.  These tests run in .NET Core 2.0 so I just tried again with the test project set to .NET Core 3.1 and saw the same correct behavior.

**Comment by jesse10581** [comment]
Ok thank you for looking.  I'll pull the source down and try debugging
myself.  The other fields come thru just fine, which is strange.

**Comment by damieng** [comment]
If you can create a minimal repro - like a console app - that shows it I can try that here.

**Comment by jesse10581** [comment]
Will do.  I'll get back to you.

**Comment by jesse10581** [comment]
I figured out the problem.  It was my fault. My fields in user_metadata that had underscores, i.e. first_name, last_name (as I am not sure how you would store those things at the root user level, but that’s another topic for another day).  Anyway, I had assumed at first that since first_name and last_name are deserialized into FirstName and LastName, that those fields would be deserialized as UserMetadata?.FirstName, UserMetadata?.LastName.  However, they, of course are not, since they are coming from a dynamic field.  So I just changed my code to reference UserMetadata?.first_name / last_name.

 

Thank you for your assistance,

 

Jesse
